### DISCUSSION TITLE
How to allow & in text input

### DISCUSSION THREAD

**Discussion by FaizPrasla** [mganss/HtmlSanitizer#430] [General]
The '&' character after sanitization gets converted to `&amp;` is there a way to allow & character without it being sanitized. Would something like `sanitizer.AllowedSchemes.Add("&")` work?

**Comment by mganss** [comment]
No, this won't work. `AllowedSchemes` specifies the allowed URI schemes, like http, ftp, mailto etc.
The ampersand character & is reserved in HTML, see https://developer.mozilla.org/en-US/docs/Glossary/Entity
The parser in AngleSharp "corrects" some malformed HTML by converting spurious characters into its equivalent entities, making the resulting HTML more valid. See also the section here: https://github.com/mganss/HtmlSanitizer#text-content-not-necessarily-preserved-as-is

**Reply by FaizPrasla** [reply]
Is there a work around where the following reserved characters &, >, <, and " can be string literals instead of being converted into their entity like &amp; Does the library support it? or is there a way to jerryrig and has been solved before?

Thank you.

**Reply by mganss** [reply]
You can replace the default formatter as described here: https://github.com/mganss/HtmlSanitizer/wiki/Examples#ex3-replacing-the-default-formatter
The relevant code in the default formatter is here: https://github.com/mganss/HtmlSanitizer/blob/5492fbedc6a2a852d231b48e567a8907a1aa2426/src/HtmlSanitizer/HtmlFormatter.cs#L58-L66

This would use `&` instead of `&amp;` for all occurrences of the ampersand character, even those that were originally entities in the input.

What's the use case?

**Reply by FaizPrasla** [reply]
Sanitizing message text field. Where the message text field is formatted as text/plain by default instead of text/html

**Reply by mganss** [reply]
If the text is not displayed as HTML, i.e. not in a browser where XSS can occur, then sanitizing shouldn't be necessary at all.

**Reply by FaizPrasla** [reply]
Text field is in a web app. So its browser submit button.

**Reply by mganss** [reply]
Where will the text be displayed again after it has been submitted?

**Reply by FaizPrasla** [reply]
Same browser in some component on the same page as where you open a modal to submit text.

**Reply by mganss** [reply]
But it's also displayed elsewhere in a non-HTML environment? Perhaps it's best to keep the original unsanitized input alongside the sanitized HTML. This way you can display the sanitized HTML in the browser and still have the original input for display in the non-HTML environment.
